kratos/python_scripts/python_solver.py

```diff
@@ -22,18 +22,24 @@ def __init__(self, model, settings):
         model -- The Model to be used
         settings -- The solver settings used
         """
-        if (type(model) != KratosMultiphysics.Model):
+        if not isinstance(model, KratosMultiphysics.Model):
             raise Exception("Input is expected to be provided as a Kratos Model object")
 
-        if (type(settings) != KratosMultiphysics.Parameters):
+        if not isinstance(settings, KratosMultiphysics.Parameters):
             raise Exception("Input is expected to be provided as a Kratos Parameters object")
 
         self.model = model
         self.settings = settings
 
         # TODO remove the check once the derived solvers implement this
         if hasattr(self, '_validate_settings_in_baseclass'):
-            self.settings.ValidateAndAssignDefaults(self.GetDefaultSettings())
+            default_settings = self.GetDefaultSettings()
+
+            default_settings.RecursivelyAddMissingParameters(self.settings)
```

Comment:
I think this is wrong, this way you will allow settings that are not in the defaults (and hence disable the validation of the parameters)

Reply:
Well, any better idea?

Reply:
not doing recursive validation ...

Reply:
I got it, on it...